stylix/testbed/default.nix

```
@@ -13,19 +14,28 @@ let
       system = lib.nixosSystem {
         inherit (pkgs) system;
 
-        modules = [
-          ./modules/common.nix
-          ./modules/enable.nix
-          ./modules/application.nix
-          inputs.self.nixosModules.stylix
-          inputs.home-manager.nixosModules.home-manager
-          testbed
+        modules =
+          [
+            ./modules/common.nix
+            ./modules/enable.nix
+            ./modules/application.nix
+            inputs.self.nixosModules.stylix
+            inputs.home-manager.nixosModules.home-manager
+            testbed
+          ]
+          ++ lib.mapAttrsToList
+            (
+              target:
+              lib.optionalAttrs (
+                lib.hasPrefix "testbed${testbedFieldSeparator}${target}${testbedFieldSeparator}" name
```

consider modules with multiple testbeds such as `cava` and `cava-rainbow`
```suggestion
                lib.hasPrefix "testbed${testbedFieldSeparator}${target}" name
```

---

> consider modules with multiple testbeds such as `cava` and `cava-rainbow`

When I thought about this, I imagined it would be better to keep the regex as restrictive as possible and explicitly opt-in:

```nix
{
  cava = /* ... */;
  cava-rainbow = /* ... */;
}
```

When to opt-in should be very easy to spot, as newly added testbeds either do not evaluate or are very occasionally non-functional at runtime. Both of which should be easy to catch.

However, on second thought, adding testbeds should be seemless and listing every allowed testbed variant here seems bad.

I will apply your suggestion.